lib/navigation/preprocess_encoders.cpp

```diff
@@ -1,13 +1,29 @@
 #include "preprocess_encoders.hpp"
 
+//#include <cmath>
+
+#include <algorithm>
+#include <numeric>
+#include <optional>
```

Comment:
Look at your includes here. Anything included in the cpp file does not need to be included in the hpp file